chain/transaction.go

```diff
@@ -275,136 +160,21 @@ func EstimateUnits(r Rules, actions Actions, authFactory AuthFactory) (fees.Dime
-		start, end := action.ValidRange(r)
-		if start >= 0 && timestamp < start {
-			return fmt.Errorf("%w: action type %d at index %d", ErrActionNotActivated, action.GetTypeID(), i)
-		}
-		if end >= 0 && timestamp > end {
-			return fmt.Errorf("%w: action type %d at index %d", ErrActionNotActivated, action.GetTypeID(), i)
-		}
-	}
-	start, end := t.Auth.ValidRange(r)
-	if start >= 0 && timestamp < start {
-		return ErrAuthNotActivated
-	}
-	if end >= 0 && timestamp > end {
-		return ErrAuthNotActivated
-	}
-	units, err := t.Units(s, r)
-	if err != nil {
-		return err
-	}
-	fee, err := feeManager.Fee(units)
-	if err != nil {
-		return err
-	}
-	return s.CanDeduct(ctx, t.Auth.Sponsor(), im, fee)
-}
-
-// Execute after knowing a transaction can pay a fee. Attempt
-// to charge the fee in as many cases as possible.
-//
-// Invariant: [PreExecute] is called just before [Execute]
-func (t *Transaction) Execute(
-	ctx context.Context,
-	feeManager *internalfees.Manager,
-	s StateManager,
-	r Rules,
-	ts *tstate.TStateView,
-	timestamp int64,
-) (*Result, error) {
-	// Always charge fee first
-	units, err := t.Units(s, r)
-	if err != nil {
-		// Should never happen
-		return nil, err
-	}
-	fee, err := feeManager.Fee(units)
-	if err != nil {
-		// Should never happen
-		return nil, err
-	}
-	if err := s.Deduct(ctx, t.Auth.Sponsor(), ts, fee); err != nil {
-		// This should never fail for low balance (as we check [CanDeductFee]
-		// immediately before).
-		return nil, err
-	}
-
-	// We create a temp state checkpoint to ensure we don't commit failed actions to state.
-	//
-	// We should favor reverting over returning an error because the caller won't be charged
-	// for a transaction that returns an error.
-	var (
-		actionStart   = ts.OpIndex()
-		actionOutputs = [][]byte{}
-	)
-	for i, action := range t.Actions {
-		actionOutput, err := action.Execute(ctx, r, ts, timestamp, t.Auth.Actor(), CreateActionID(t.ID(), uint8(i)))
-		if err != nil {
-			ts.Rollback(ctx, actionStart)
-			return &Result{false, utils.ErrBytes(err), actionOutputs, units, fee}, nil
-		}
-
-		var encodedOutput []byte
-		if actionOutput == nil {
-			// Ensure output standardization (match form we will
-			// unmarshal)
-			encodedOutput = []byte{}
-		} else {
-			encodedOutput, err = MarshalTyped(actionOutput)
-			if err != nil {
-				return nil, err
-			}
-		}
-
-		actionOutputs = append(actionOutputs, encodedOutput)
-	}
-	return &Result{
-		Success: true,
-		Error:   []byte{},
-
-		Outputs: actionOutputs,
-
-		Units: units,
-		Fee:   fee,
-	}, nil
-}
-
 func (t *Transaction) Marshal(p *codec.Packer) error {
 	if len(t.bytes) > 0 {
 		p.PackFixedBytes(t.bytes)
 		return p.Err()
 	}
-	return t.marshal(p, true)
+	return t.marshal(p)
 }
 
-func (t *Transaction) marshal(p *codec.Packer, marshalSignature bool) error {
+func (t *Transaction) marshal(p *codec.Packer) error {
 	t.Base.Marshal(p)
-	if err := t.Actions.marshalInto(p); err != nil {
+	if err := p.Err(); err != nil {
```

Comment:
I think this diff isn't needed since `Packer` will stop at the first error ([ref](https://github.com/ava-labs/avalanchego/blob/0c4efd743e1d737f4e8970d0e0ebf229ea44406c/utils/wrappers/errors.go#L13)).

Reply:
I see.. so you're saying that adding this would have no practical functional impact since the packet would effectively be "blocked" for any new additions due to the error.